### ISSUE TITLE
Question about matcher

### ISSUE THREAD

**Issue by rachitagarwal2603** [ZrrSkywalker/MonoDETR#10]
First of all, it's a great approach. Thank you for that.
I have some doubts in Matcher implementation. I am not able to figure about how you are neglecting dummy values appended to the target boxes and labels during loss calculation.
To elaborate more with an example, suppose for an image there 3 target GT boxes, and max objs=50(let). So in this case remaining 47 values will be dummy right(all zeros). Now in matcher and later in loss calc, all these 50 object values are used, irrespective of whether it's dummy or not. But logically loss penalty should be based on 3 GT boxes only right.
Why is that? Am I missing something here? 
Thank you in advance and sorry for big explanation :)

**Comment by rachitagarwal2603** [comment]
Also can you please share the pretrained model.

**Comment by ZrrSkywalker** [comment]
Thanks for your interest!
We refer to the official code of Deformable DETR and generate an 'indices' variable to index the valid queries. During loss functions, we first use this 'indices' to filter out the valid queries and only calculate the loss for these queries.
